frame/contracts/src/lib.rs

```diff
@@ -319,14 +319,17 @@ pub mod pallet {
 		/// The maximum length of the debug buffer in bytes.
 		#[pallet::constant]
 		type MaxDebugBufferLen: Get<u32>;
+
+		/// The sequence of migrations applied.
+		type Migrations: MigrateSequence;
 	}
 
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
 		fn on_idle(_block: T::BlockNumber, remaining_weight: Weight) -> Weight {
 			use migration::MigrateResult::*;
 
-			let (result, weight) = Migration::<T>::migrate(remaining_weight);
+			let (result, weight) = Migration::<T, T::Migrations>::migrate(remaining_weight);
```

Comment:
since the first generic is T we might be able to infer the second one directly instead of specifying it